server/src/main/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderator.java

```diff
@@ -527,8 +547,7 @@ private Sink getOrCreateSink(final SegmentIdWithShardSpec identifier)
            .emit();
       }
 
-      sinks.put(identifier, retVal);
-      metrics.setSinkCount(sinks.size());
+      addSink(identifier, retVal);
       sinkTimeline.add(retVal.getInterval(), retVal.getVersion(), identifier.getShardSpec().createChunk(retVal));
```

Comment:
This line is already moved inside `addSink()` method. Should it be removed from here?

Reply:
Removed. Thanks